packages/cli/src/lib/bundler/config.ts

```diff
@@ -70,27 +70,74 @@ async function readBuildInfo() {
   };
 }
 
-async function loadLernaPackages(): Promise<
-  { name: string; location: string }[]
-> {
+type LernaPackage = {
+  name: string;
+  location: string;
+};
+
+async function loadLernaPackages(): Promise<LernaPackage[]> {
   const LernaProject = require('@lerna/project');
   const project = new LernaProject(cliPaths.targetDir);
   return project.getPackages();
 }
 
+// Enables proper resolution of packages when linking in external packages.
+// Without this the packages would depend on dependencies in the node_modules
+// of the external packages themselves, leading to module duplication
+class LinkedPackageResolvePlugin implements ResolvePlugin {
+  constructor(
+    private readonly targetModules: string,
+    private readonly packages: LernaPackage[],
+  ) {}
+
+  apply(resolver: any /* EnhancedResolve.Resolver */) {
+    resolver.hooks.resolve.tapAsync(
+      'LinkedPackageResolvePlugin',
+      (request: any, context: any, callback: () => void) => {
+        const pkg = this.packages.find(pkg =>
+          request.path.startsWith(pkg.location),
```

Comment:
Gotta be honest that I'd have to read up on webpack to reliably review the code below. I guess a comment here and there with example values of what could be the value of the variables, could clarify a bit.

Reply:
Added a bit more explanation